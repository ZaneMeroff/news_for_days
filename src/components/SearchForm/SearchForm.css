
.search-form-container {
  align-items: center;
  background-color: rgb(232, 90, 90, 1);
  display: flex;
  flex-direction: row;
  height: 60px;
  margin-top: -60px;
  padding-left: 5%;
  position: fixed;
  width: 100%;
}

.search-input {
  color: #000000;
  border-radius: 15px;
  height: 30px;
  padding-left: 1%;
  width: 30%;
}

.search-button {
  border-radius: 15px;
  color: #000000;
  height: 30px;
  margin-left: 4%;
  text-transform: uppercase;
  width: 10%;
}

.search-button:hover {
  cursor: pointer;
}
