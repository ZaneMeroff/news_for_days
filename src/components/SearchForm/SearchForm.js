import React, { Component } from 'react';
import './SearchForm.css';

class SearchForm extends Component {
  constructor() {
    super();
    this.state = {};
  }

  onSearchChange = event => {
    this.setState({query: event.target.value})
  }

  render() {
    return (
    <section className='search-form-container'>
     <input className='search-input' onChange={this.onSearchChange} placeholder='search for articles here...'></input>
     <button onClick={() => this.props.filterSearchResults(this.state.query)} className='search-button'>search</button>
    </section>
    )
  }

}

export default SearchForm;
