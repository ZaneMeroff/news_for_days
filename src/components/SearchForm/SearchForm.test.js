import React from 'react';
import SearchForm from '../SearchForm/SearchForm';
import { shallow } from 'enzyme';

describe('SearchForm', () => {

  it('should be able to render SearchForm to page', () => {
    const wrapper = shallow(<SearchForm
    filterSearchResults={jest.fn()}
    />)
    expect(wrapper).toMatchSnapshot();
  });

  it('expect filterSearchResults to be called on button click', () => {
    const mockFilterSearchResults = jest.fn();
    const wrapper = shallow(<SearchForm
    filterSearchResults={mockFilterSearchResults}
    />)
    wrapper.find('button').simulate('click')
    expect(mockFilterSearchResults).toHaveBeenCalled();
  });

  it('should update state when onSearchChange is called', () => {
    const wrapper = shallow(<SearchForm
    filterSearchResults={jest.fn()}
    />)
    const mockEvent = {
      target: {
        value: 'chicken'
      }
    }
    wrapper.instance().onSearchChange(mockEvent)
    expect(wrapper.state('query')).toEqual('chicken')
  });

});
