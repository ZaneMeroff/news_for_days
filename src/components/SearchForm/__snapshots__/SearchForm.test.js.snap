// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should be able to render SearchForm to page 1`] = `
<section
  className="search-form-container"
>
  <input
    className="search-input"
    onChange={[Function]}
    placeholder="search for articles here..."
  />
  <button
    className="search-button"
    onClick={[Function]}
  >
    search
  </button>
</section>
`;
